src/stores/SetupEncryptionStore.js

```diff
@@ -49,6 +49,15 @@ export class SetupEncryptionStore extends EventEmitter {
         MatrixClientPeg.get().on("crypto.verification.request", this.onVerificationRequest);
         MatrixClientPeg.get().on('userTrustStatusChanged', this._onUserTrustStatusChanged);
 
+        const cli = MatrixClientPeg.get();
+        const requestsInProgress = cli.getVerificationRequestsToDeviceInProgress(cli.getUserId());
+        if (requestsInProgress.length) {
+            // If there are multiple, we take the first. Equally if the user sends another request from
+            // another device after this screen has been shown, we'll switch to the new one, so this
+            // generally doesn't support multiple requests.
+            this._setActiveVerificationRequest(requestsInProgress[0]);
```

Comment:
I'm assuming without checking that the JS SDK actually does append rather than prepend subsequent requests... 😅

Reply:
Actually I was assuming they'd be in no particular order, but ES6 maps are ordered so the most recent should be the last, although it's probably all much the same, we may as well try to be consistent I guess.